File: example-java/src/main/java/com/github/simonpercic/oklogexamplejava/DataFetcher3.java
package com.github.simonpercic.oklogexamplejava;

import com.github.simonpercic.oklogexamplejava.data.api.okhttp3.DataModule3;
import com.github.simonpercic.oklogexamplejava.data.api.okhttp3.RestApi3;
import com.github.simonpercic.oklogexamplejava.data.api.model.response.show.ShowResponse;

import java.io.IOException;
import java.util.List;

import javax.inject.Inject;

import retrofit2.Response;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
class DataFetcher3 {

    @Inject RestApi3 restApi;

    void fetch() {
        DaggerDataFetcher3Component.builder()
                .dataModule3(new DataModule3())
                .build()
                .inject(this);

        try {
            Response<List<ShowResponse>> response = restApi.getShows().execute();

            if (response.isSuccessful()) {
                System.out.printf("Got %d shows%n", response.body().size());
            } else {
               System.out.println("Call failed");
            }
        } catch (IOException e) {
            System.out.printf("Call failed with %s%n", e.getMessage());
        }
    }
}

File: example-kotlin/src/main/kotlin/com/github/simonpercic/oklogexamplekotlin/DataFetcher3.kt
package com.github.simonpercic.oklogexamplekotlin

import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp3.DataModule3
import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp3.RestApi3
import java.io.IOException
import javax.inject.Inject

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
class DataFetcher3 {

    @Inject
    lateinit var restApi: RestApi3

    fun fetch() {
        DaggerDataFetcher3Component.builder()
            .dataModule3(DataModule3())
            .build()
            .inject(this)

        try {
            val response = restApi.getShows().execute()

            if (response.isSuccessful) {
                System.out.printf("Got %d shows%n", response.body()!!.size)
            } else {
                println("Call failed")
            }
        } catch (e: IOException) {
            System.out.printf("Call failed with %s%n", e.message)
        }

    }
}
